### PR TITLE
build+lnrpc: bump gRPC + protobuf related dependencies to latest version

### LINKED ISSUE #1699: Make RPC fails with error code
I followed the instructions on  https://github.com/lightningnetwork/lnd/tree/master/lnrpc  and get this error:

home/lightningmaster/protoc/protoc/bin/protoc: 1: /home/lightningmaster/protoc/protoc/bin/protoc: Syntax error: "(" unexpected

This is using the rpc.proto that's at the current commit level.

### PR DESCRIPTION
**Pull request by wpaulino**
Went through the current docs on my new setup and realized some deps were missing, so we now explicitly use `go get`.

Fixes #1699.

### CODE REVIEW

File: lnrpc/README.md [hunk 1]
```diff
@@ -151,17 +151,18 @@ $ export PATH=$PWD/protoc/bin:$PATH
 
 2. Install `golang/protobuf` at commit `ab9f9a6dab164b7d1246e0e688b0ab7b94d8553e`.
 ```bash
-$ git clone https://github.com/golang/protobuf $GOPATH/src/github.com/golang/protobuf
+$ go get -u github.com/golang/protobuf
```
Comment (ZapUser77) [inline lnrpc/README.md]: This command yields the following error: 
package github.com/golang/protobuf: no Go files in /home/l/gocode/src/github.com/golang/protobuf

Comment (wpaulino) [reply lnrpc/README.md]: Yep should be `github.com/golang/protobuf/protoc-gen-go`. Fixed!

**Comment by halseth** [comment]
Should be consolidated with https://github.com/lightningnetwork/lnd/pull/1467

**Comment by ZapUser77** [comment]
Ok, solved the problem.
Took your instructions, and added one additional step.
After "Make install" on line 157, I added "Checkout master".
This seemed to be the critical step needed to get everything working.

Thanks for the help.

**Comment by wpaulino** [comment]
This PR has been reworked to replace #1467. The `rpc.proto` file had to be broken down into two, each containing a single service. This is needed to prevent the following [diff](https://gist.github.com/wpaulino/16cb7030ed8606ace8905b4beec4f8c9), where the top 4 RPCs of the `Lightning` service would be replaced with the `WalletUnlocker` RPCs.

**Comment by wpaulino** [comment]
Before landing this in, we should also update the documentation parser (https://api.lightning.community) to reflect the new changes.

File: lnrpc/README.md [hunk 2]
```diff
@@ -140,36 +140,35 @@ $ go get -u github.com/lightningnetwork/lnd/lnrpc
...
 ```bash
-$ go get google.golang.org/genproto
-$ cd $GOPATH/src/google.golang.org/genproto
-$ git reset --hard a8101f21cf983e773d0c1133ebc5424792003214
+$ go get -d google.golang.org/genproto
+$ git -C $GOPATH/src/google.golang.org/genproto checkout bd91e49a0898e27abb88c339b432fa53d7497ac0
 ```
 
-4. Install `grpc-ecosystem/grpc-gateway` at commit `f2862b476edcef83412c7af8687c9cd8e4097c0f`.
+4. Install v1.5.1 of `grpc-ecosystem/grpc-gateway`.
```
Comment (halseth) [inline lnrpc/README.md]: why not `v1.6.2`, which is latest?

**Pull request closed without merging**